docs/2.guide/2.directory-structure/1.pages.md

```diff
@@ -266,17 +266,24 @@ console.log(route.meta.title) // My home page
 
 If you are using nested routes, the page metadata from all these routes will be merged into a single object. For more on route meta, see the [vue-router docs](https://router.vuejs.org/guide/advanced/meta.html#route-meta-fields).
 
-Much like `defineEmits` or `defineProps` (see [Vue docs](https://vuejs.org/api/sfc-script-setup.html#defineprops-defineemits)), `definePageMeta` is a **compiler macro**. It will be compiled away so you cannot reference it within your component. Instead, the metadata passed to it will be hoisted out of the component. Therefore, the page meta object cannot reference the component (or values defined on the component). However, it can reference imported bindings.
+Much like `defineEmits` or `defineProps` (see [Vue docs](https://vuejs.org/api/sfc-script-setup.html#defineprops-defineemits)), `definePageMeta` is a **compiler macro**. It will be compiled away so you cannot reference it within your component. Instead, the metadata passed to it will be hoisted out of the component.
+Therefore, the page meta object cannot reference the component. However, it can reference imported bindings, as well as locally defined **pure functions**.
+
+::warning
+Make sure not to reference any reactive data or functions that cause side effects. This can lead to unexpected behavior.
+::
 
 ```vue
 <script setup lang="ts">
 import { someData } from '~/utils/example'
 
-const title = ref('')
+function validateIdParam(route) {
+  return route.params.id && !isNaN(Number(route.params.id))
+}
 
 definePageMeta({
-  title,  // This will create an error
```

Comment:
I think it's worth preserving, as this example is a warning example...

Reply:
It won't create an error now, though. The ref will be extracted too
```ts
const title = ref("");
const __nuxt_page_meta = {
      title
    }
export default __nuxt_page_meta
```

It doesn't really make sense to use a ref in this way, since it will be a different ref than the one in the component - hence the warning "Make sure not to reference any reactive data"

I've added it back with a different comment

Reply:
ah, of course.

thank you 🙏